js/ui/screenShield.js

```diff
@@ -640,6 +640,9 @@ var ScreenShield = class {
             if (Main.sessionMode.currentMode !== 'unlock-dialog-payg' &&
                 Main.paygManager.isLocked)
                 Main.sessionMode.pushMode('unlock-dialog-payg');
+
+            this._clearCurrentDialog();
+            this._ensureUnlockDialog(true);
```

Comment:
A couple things:
- I think we should probably call this only if we are changing the mode in either conditionals above, otherwise every time we lock the screen in the "normal" case (without payg) we would be deleting and recreating the dialog? Also should we check whether `_ensureUnlockDialog()` fails as done above at the start of the method?
- We should also probably move the call to `_ensureUnlockDialog()` at the start of the method to the "else" clause here (when `this._isGreeter` is true).

Btw, the commit msg adding this change says `The reason is that in screenShield.animate()` and I guess you meant `screenShield.activate()` here?

Reply:
> A couple things:
> 
>     * I think we should probably call this only if we are changing the mode in either conditionals above, otherwise every time we lock the screen in the "normal" case (without payg) we would be deleting and recreating the dialog?

You're right, I moved it up into the if block.

>Also should we check whether `_ensureUnlockDialog()` fails as done above at the start of the method?

No, because we know the unlock-dialog-payg session mode supports unlocking, so there's no chance it will fail.
> 
>     * We should also probably move the call to `_ensureUnlockDialog()` at the start of the method to the "else" clause here (when `this._isGreeter` is true).
No, I don't think so, because that would regress T29527 (initial setup authentication error). We need the early return for session modes that don't support locking.
> 
> 
> Btw, the commit msg adding this change says `The reason is that in screenShield.animate()` and I guess you meant `screenShield.activate()` here?

Yes, that was a mistake.